qiskit_optimization/applications/max_k_cut.py

```diff
@@ -0,0 +1,247 @@
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""An application class for the Max-k-cut."""
+
+from typing import List, Dict, Tuple, Optional, Union
+import networkx as nx
+import numpy as np
+from docplex.mp.model import Model
+
+from qiskit.utils import algorithm_globals
+from qiskit_optimization.algorithms import OptimizationResult
+from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+from qiskit_optimization.translators import from_docplex_mp
+from .graph_optimization_application import GraphOptimizationApplication
+
+try:
+    from matplotlib.pyplot import cm
+    from matplotlib.colors import to_rgba
+
+    _HAS_MATPLOTLIB = True
+except ImportError:
+
+    _HAS_MATPLOTLIB = False
+
+
+class Maxkcut(GraphOptimizationApplication):
+    """Optimization application for the "max-k-cut" [1] problem based on a NetworkX graph.
+
+    References:
+        [1]: Z. Tabi et al.,
+             "Quantum Optimization for the Graph Coloring Problem with Space-Efficient Embedding"
+             2020 IEEE International Conference on Quantum Computing and Engineering (QCE),
+             2020, pp. 56-62, doi: 10.1109/QCE49297.2020.00018.,
+             https://ieeexplore.ieee.org/document/9259934
+    """
+
+    def __init__(
+        self,
+        graph: Union[nx.Graph, np.ndarray, List],
+        k: int,
+    ) -> None:
+        """
+        Args:
+            graph: A graph representing a problem. It can be specified directly as a
+                `NetworkX <https://networkx.org/>`_ graph,
+                or as an array or list format suitable to build out a NetworkX graph.
+            k: The number of colors
+        """
+        super().__init__(graph=graph)
+        self._subsets_num = k
+        self._colors: Union[List[Tuple[float, float, float, float]], List[str]] = None
+        self._seed: int = None
+
+    def to_quadratic_program(self) -> QuadraticProgram:
+        """Convert a Max-k-cut problem instance into a
+        :class:`~qiskit_optimization.problems.QuadraticProgram`
+
+        Returns:
+            The :class:`~qiskit_optimization.problems.QuadraticProgram` created
+            from the Max-k-cut problem instance.
+        """
+        for w, v in self._graph.edges:
+            self._graph.edges[w, v].setdefault("weight", 1)
+
+        mdl = Model(name="Max-k-cut")
+        n = self._graph.number_of_nodes()
+        k = self._subsets_num
+        x = {(v, i): mdl.binary_var(name=f"x_{v}_{i}") for v in range(n) for i in range(k)}
+        first_penalty = mdl.sum_squares((1 - mdl.sum(x[v, i] for i in range(k)) for v in range(n)))
```

Comment:
Could you make a model with a standard formulation of the problem? Please do not penalize constraints by yourself.
In the application classes, we make a model with a standard formulation. The conversion to QUBO will be done within algorithm classes.